File: sol/solution/0001-0100/0001/Solution.dart
class Solution {
    List<int> twoSum(List<int> nums, int target) {
        Map<int, int> map = {};
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                return [map[target - nums[i]]!, i];
            }
            map[nums[i]] = i;
        }
        return [];
    }
}

File: sol/solution/0001-0100/0001/Solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut d = HashMap::new();
        for (i, &x) in nums.iter().enumerate() {
            if let Some(&j) = d.get(&(target - x)) {
                return vec![j as i32, i as i32];
            }
            d.insert(x, i);
        }
        vec![]
    }
}